modules/apps/27-interchain-accounts/controller/keeper/handshake.go

```diff
@@ -28,26 +28,35 @@ func (k Keeper) OnChanOpenInit(
 	chanCap *capabilitytypes.Capability,
 	counterparty channeltypes.Counterparty,
 	version string,
-) error {
+) (string, error) {
 	if order != channeltypes.ORDERED {
-		return sdkerrors.Wrapf(channeltypes.ErrInvalidChannelOrdering, "expected %s channel, got %s", channeltypes.ORDERED, order)
+		return "", sdkerrors.Wrapf(channeltypes.ErrInvalidChannelOrdering, "expected %s channel, got %s", channeltypes.ORDERED, order)
 	}
 
 	if !strings.HasPrefix(portID, icatypes.PortPrefix) {
-		return sdkerrors.Wrapf(icatypes.ErrInvalidControllerPort, "expected %s{owner-account-address}, got %s", icatypes.PortPrefix, portID)
+		return "", sdkerrors.Wrapf(icatypes.ErrInvalidControllerPort, "expected %s{owner-account-address}, got %s", icatypes.PortPrefix, portID)
 	}
 
 	if counterparty.PortId != icatypes.PortID {
-		return sdkerrors.Wrapf(icatypes.ErrInvalidHostPort, "expected %s, got %s", icatypes.PortID, counterparty.PortId)
+		return "", sdkerrors.Wrapf(icatypes.ErrInvalidHostPort, "expected %s, got %s", icatypes.PortID, counterparty.PortId)
 	}
 
 	var metadata icatypes.Metadata
-	if err := icatypes.ModuleCdc.UnmarshalJSON([]byte(version), &metadata); err != nil {
-		return sdkerrors.Wrapf(icatypes.ErrUnknownDataType, "cannot unmarshal ICS-27 interchain accounts metadata")
+	if strings.TrimSpace(version) == "" {
+		connection, err := k.channelKeeper.GetConnection(ctx, connectionHops[0])
+		if err != nil {
```

Comment:
just a super tiny nit but you have the err statement in the second else clause in the  `; err != nil` setup, would be nice if they were both like this or both the other way just as a tiny remark on style

Reply:
But we need to use the `connection` below outside the scope of the returned error.
Generally we use the below for function return sigs with one single `error` return:
```go
if err := invokeFunc(); err != nil {
    // handle err
}
```

If we use the following then we can't use the connection below, outside the scope of the conditional:
```go
if connection, err := k.channelKeeper.GetConnection(ctx, connectionHops[0]); err != nil {
   // connection and err are only available at this level of scope
}
```

Reply:
you should technically be able to if you put the usage in an else clause, but I personally prefer the way you have it now rather than adding additional `else` clauses!